File: app/src/main/java/org/ametro/render/elements/BitmapBackgroundElement.java
package org.ametro.render.elements;

import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Paint;
import android.graphics.Rect;

import org.ametro.model.entities.MapScheme;
import org.ametro.render.RenderConstants;

public class BitmapBackgroundElement extends DrawingElement {

    private final Paint paint;
    private final Bitmap bitmap;

    public BitmapBackgroundElement(MapScheme scheme, Bitmap bitmap) {
        super();
        this.bitmap = bitmap;
        paint = new Paint();
        setBoxAndPriority(new Rect(0, 0, scheme.getWidth(), scheme.getHeight()), RenderConstants.TYPE_BACKGROUND);
    }

    @Override
    public void draw(Canvas canvas) {
        canvas.drawBitmap(bitmap, 0, 0, paint);
    }
}

File: app/src/main/java/io/github/romangolovanov/apps/ametro/render/elements/BitmapBackgroundElement.kt
package io.github.romangolovanov.apps.ametro.render.elements

import android.graphics.Bitmap
import android.graphics.Canvas
import android.graphics.Paint
import android.graphics.Rect
import io.github.romangolovanov.apps.ametro.model.entities.MapScheme
import io.github.romangolovanov.apps.ametro.render.RenderConstants

class BitmapBackgroundElement(scheme: MapScheme, private val bitmap: Bitmap) : DrawingElement() {

    private val paint = Paint()

    init {
        setBoxAndPriority(Rect(0, 0, scheme.width, scheme.height), RenderConstants.TYPE_BACKGROUND)
    }

    override fun draw(canvas: Canvas) {
        canvas.drawBitmap(bitmap, 0f, 0f, paint)
    }
}
